### PR TITLE
iOS FFI exploration

### PR DESCRIPTION
**Pull request by ivnsch**
- Passing struct as parameter from app
- Returning struct to app
- Passing callback as parameter from app

Note: This contaminates a little the current headers, but it seems fine for now.

### CODE REVIEW

File: src/ios/ffi_for_sanity_tests.rs [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+use core_foundation::{base::TCFType, string::{CFString, CFStringRef}};
+use libc::c_char;
+use std::{thread, sync::mpsc::{self, Sender}};
+use mpsc::Receiver;
+
+// Expose an interface for apps (for now only iOS) to test that general FFI is working as expected.
+// i.e. assumptions on which the actual FFI interface relies.
+// TODO can the c headers for this be generated in a separate file? Needs adjustments in script to generate lib and framework too.
+
+#[repr(C)]
```
Comment (duskoo) [inline src/ios/ffi_for_sanity_tests.rs:12]: https://doc.rust-lang.org/reference/type-layout.html#reprc-structs

File: src/ios/ffi_for_sanity_tests.rs [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+  let my_struct = MyStruct {
+    my_int: (*par).my_int,
+    my_str: my_str.to_owned(),
+    my_u8: (*par).my_nested.my_u8,
+  };
+
+  println!("Received struct from iOS: {:?}", my_struct);
+
+  1
```
Comment (duskoo) [inline src/ios/ffi_for_sanity_tests.rs:43]: @i-schuetz why is it required to return i32? Void not supported?

**Review by duskoo — APPROVED** [review]
@i-schuetz Any good docs/blags to help understanding FFI stuff?
Found this: https://doc.rust-lang.org/nomicon/ffi.html

**Comment by ivnsch** [comment]
https://doc.rust-lang.org/1.9.0/book/ffi.html
https://michael-f-bryan.github.io/rust-ffi-guide/
https://www.youtube.com/watch?v=5IJr7wn48_I (German)

File: src/ios/ffi_for_sanity_tests.rs (hunk 2 above)
Comment (ivnsch) [reply src/ios/ffi_for_sanity_tests.rs:43]: Just returning something to double-check in the unit tests that it's terminating correctly.

**Pull request merged**